src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -317,18 +335,20 @@ public Object getAtOffsetULong(DynamicObject pointer, int offset, int type) {
 
         @Specialization(guards = "type == TYPE_LL")
         public long getAtOffsetLL(DynamicObject pointer, int offset, int type) {
-            return Layouts.POINTER.getPointer(pointer).getLongLong(offset);
+            return PointerOperations.getLongLong(Layouts.POINTER.getPointer(pointer), offset);
         }
 
         @Specialization(guards = "type == TYPE_ULL")
         public Object getAtOffsetULL(DynamicObject pointer, int offset, int type) {
             return PointerReadLongPrimitiveNode.readUnsignedLong(getContext(), pointer, offset);
         }
 
-        @TruffleBoundary
         @Specialization(guards = "type == TYPE_STRING")
         public DynamicObject getAtOffsetString(DynamicObject pointer, int offset, int type) {
-            return createString(StringOperations.encodeRope(Layouts.POINTER.getPointer(pointer).getString(offset), UTF8Encoding.INSTANCE));
+            return createString(
+                    StringOperations.encodeRope(
+                            PointerOperations.getString(Layouts.POINTER.getPointer(pointer), offset),
```

Comment:
An aside: I wonder if Pointer#getString is the same as the logic above for readStringToNull().